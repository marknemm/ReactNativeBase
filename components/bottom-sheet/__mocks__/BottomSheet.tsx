import type { BottomSheetProps } from '@components/bottom-sheet/BottomSheet.interfaces';

const MockBottomSheet = jest.fn().mockImplementation((props: BottomSheetProps) => {
  const ActualComponent = jest.requireActual('@components/bottom-sheet/BottomSheet').default;
  return <ActualComponent {...props} />;
});

export default MockBottomSheet;
