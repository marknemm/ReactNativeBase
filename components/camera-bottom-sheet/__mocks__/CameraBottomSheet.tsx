import type { CameraBottomSheetProps } from '@components/camera-bottom-sheet/CameraBottomSheet.interfaces';

const MockCameraBottomSheet = jest.fn().mockImplementation((props: CameraBottomSheetProps) => {
  const ActualComponent = jest.requireActual('@components/camera-bottom-sheet/CameraBottomSheet').default;
  return <ActualComponent {...props} />;
});

export default MockCameraBottomSheet;
