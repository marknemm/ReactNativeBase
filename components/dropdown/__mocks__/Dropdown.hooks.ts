/**
 * Mocks the useDropdownItems hook.
 */
export const useDropdownItems = jest.fn().mockImplementation((data) => data);
