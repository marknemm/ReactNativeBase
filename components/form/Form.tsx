import { FormProvider } from 'react-hook-form';
import { View } from 'react-native';
import type { FormProps } from './Form.interface';

/**
 * A component that wraps a form and provides form context to its children.
 *
 * @param props The component {@link FormProps}.
 * @returns The {@link Form} component.
 */
const Form: React.FC<FormProps> = ({ children, form, style, ...restViewProps }) => (
  <FormProvider {...form}>
    <View
      style={style}
      testID="rnb-form-view"
      {...restViewProps}
    >
      {children}
    </View>
  </FormProvider>
);

export type * from './Form.interface';
export default Form;
