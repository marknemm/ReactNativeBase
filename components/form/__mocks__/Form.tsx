import type { FormProps } from '@components/form/Form.interface';

const FormMock = jest.fn().mockImplementation((props: FormProps) => {
  const ActualComponent = jest.requireActual('@components/form/Form').default;
  return <ActualComponent {...props} />;
});

export default FormMock;
