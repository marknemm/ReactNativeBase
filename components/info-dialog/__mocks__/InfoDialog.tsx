import type { InfoDialogProps } from '@components/info-dialog/InfoDialog.interfaces';

const MockInfoDialog = jest.fn().mockImplementation((props: InfoDialogProps) => {
  const ActualComponent = jest.requireActual('@components/info-dialog/InfoDialog').default;
  return <ActualComponent {...props} />;
});

export default MockInfoDialog;
