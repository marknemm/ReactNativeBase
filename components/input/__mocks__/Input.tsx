import type { InputProps } from '@components/input/Input.interfaces';

const MockInput = jest.fn().mockImplementation((props: InputProps) => {
  const ActualComponent = jest.requireActual('@components/input/Input').default;
  return <ActualComponent {...props} />;
});

export default MockInput;
