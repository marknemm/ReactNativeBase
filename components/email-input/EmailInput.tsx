import Input from '@components/input/Input';
import { EMAIL_PATTERN_RULE } from '@constants/validation';
import { forwardRef } from 'react';
import type { EmailInputFC, EmailInputProps } from './EmailInput.interfaces';

/**
 * An {@link Input} field for entering an email address.
 *
 * @param props The component {@link EmailInputProps}.
 * @param ref The component reference.
 * @returns The {@link EmailInput} component.
 */
const EmailInput: EmailInputFC = forwardRef((props, ref) => (
  <Input
    autoCapitalize="none"
    autoComplete="email"
    autoCorrect={false}
    keyboardType="email-address"
    pattern={EMAIL_PATTERN_RULE}
    ref={ref}
    textContentType="emailAddress"
    {...props}
  />
));

export type * from './EmailInput.interfaces';
export default EmailInput;
