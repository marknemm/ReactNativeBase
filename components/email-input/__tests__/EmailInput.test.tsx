import EmailInput from '@components/email-input/EmailInput';
import Form from '@components/form/Form';
import { EMAIL_PATTERN_RULE } from '@constants/validation';
import { genMockForm } from '@hooks/__mocks__/form-hooks';
import AppProvider from '@test/contexts/app/AppProvider';
import { act, render, screen, userEvent } from '@testing-library/react-native';
import { cloneElement } from 'react';

describe('<EmailInput />', () => {
  describe('validation', () => {
    it('should show validation error with invalid email', async () => {
      const form = genMockForm({
        emailInput: '',
      });

      const renderJSX = (
        <Form form={form}>
          <EmailInput name="emailInput" />
        </Form>
      );

      render(
        renderJSX,
        { wrapper: AppProvider }
      );

      const textInput = screen.getByTestId('RNE__Input__text-input');
      await userEvent.type(textInput, 'testuser@domain');

      await act(() =>
        form.handleSubmit(() => {})()
      );

      expect(form.formState.errors.emailInput.type).toBe('pattern');

      screen.rerender(
        cloneElement(renderJSX, { form })
      );

      const errorText = screen.getByText(EMAIL_PATTERN_RULE.message);
      expect(errorText).toBeVisible();
    });

    it('should not show validation error with valid email', async () => {
      const form = genMockForm({
        emailInput: '',
      });

      const renderJSX = (
        <Form form={form}>
          <EmailInput name="emailInput" />
        </Form>
      );

      render(
        renderJSX,
        { wrapper: AppProvider }
      );

      const textInput = screen.getByTestId('RNE__Input__text-input');
      await userEvent.type(textInput, 'testuser@domain.com');

      await act(() =>
        form.handleSubmit(() => {})()
      );

      expect(form.formState.errors.emailInput).toBeUndefined();
    });
  });

  describe('snapshots', () => {
    it('renders correctly', () => {
      render(
        <EmailInput />,
        { wrapper: AppProvider }
      );

      expect(screen.toJSON()).toMatchSnapshot();
    });
  });
});
