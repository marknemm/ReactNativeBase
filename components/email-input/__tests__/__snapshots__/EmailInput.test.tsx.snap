// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmailInput /> snapshots renders correctly 1`] = `
<View
  style={
    {
      "marginBottom": 6.285714285714286,
      "paddingHorizontal": 0,
      "width": "100%",
    }
  }
  testID="RNE__Input__view-wrapper"
>
  <View
    collapsable={false}
    style={
      {
        "alignItems": "center",
        "borderBottomColor": "#bbb",
        "borderBottomWidth": 1,
        "borderColor": "#86939e",
        "flexDirection": "row",
        "height": 98.9730612244898,
        "transform": [
          {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <TextInput
      autoCapitalize="none"
      autoComplete="email"
      autoCorrect={false}
      editable={true}
      keyboardAppearance="light"
      keyboardType="email-address"
      onChangeText={[Function]}
      pattern={
        {
          "message": "Invalid email address",
          "value": /\\^\\[\\^\\\\s@\\]\\+@\\[\\^\\\\s@\\]\\+\\\\\\.\\[\\^\\\\s@\\]\\+\\$/,
        }
      }
      placeholderTextColor="#86939e"
      style={
        {
          "color": "#242424",
          "flex": 1,
          "fontSize": 23.57142857142857,
          "height": 98.9730612244898,
          "minHeight": 40,
        }
      }
      testID="RNE__Input__text-input"
      textContentType="emailAddress"
      underlineColorAndroid="transparent"
    />
  </View>
  <Text
    style={
      {
        "color": "#ff3b30",
        "fontSize": 12,
        "margin": 5,
      }
    }
  />
</View>
`;
